lib/main.dart

```diff
@@ -78,41 +87,40 @@ Future<void> createIconsFromArguments(List<String> arguments) async {
   }
 }
 
-Future<void> createIconsFromConfig(Map<String, dynamic> config, [String flavor]) async {
-  if (!isImagePathInConfig(config)) {
-    throw const InvalidConfigException(errorMissingImagePath);
-  }
+Future<void> createIconsFromConfig(Map<String, dynamic> config,
+    [String flavor]) async {
   if (!hasPlatformConfig(config)) {
     throw const InvalidConfigException(errorMissingPlatform);
   }
 
-  if (isNeedingNewAndroidIcon(config) || hasAndroidAdaptiveConfig(config)) {
-    final int minSdk = android_launcher_icons.minSdk();
-    if (minSdk < 26 &&
-        hasAndroidAdaptiveConfig(config) &&
-        !hasAndroidConfig(config)) {
-      throw const InvalidConfigException(errorMissingRegularAndroid);
-    }
-  }
+  for (final AbstractPlatform platform in platforms.values) {
+    final String complaint = platform.isConfigValid(config);
 
-  if (isNeedingNewAndroidIcon(config)) {
-    android_launcher_icons.createDefaultIcons(config, flavor);
-  }
-  if (hasAndroidAdaptiveConfig(config)) {
-    android_launcher_icons.createAdaptiveIcons(config, flavor);
+    if (complaint != null) {
+      throw InvalidConfigException(complaint);
+    }
   }
-  if (isNeedingNewIOSIcon(config)) {
-    ios_launcher_icons.createIcons(config, flavor);
 
+  for (final AbstractPlatform platform in platforms.values) {
+    if (platform.inConfig(config) &&
+        platform.logWarnings(config, out: stderr)) {
+      platform.createIcons(config, flavor);
+    }
   }
 }
 
 Map<String, dynamic> loadConfigFileFromArgResults(ArgResults argResults,
-    {bool verbose}) {
+    {bool verbose, String cwd}) {
   verbose ??= false;
-  final String configFile = argResults[fileOption];
+  cwd ??= './';
+
+  String configFile = argResults[fileOption];
   final String fileOptionResult = argResults[fileOption];
 
+  if (configFile != null) {
+    configFile = path.join(cwd, configFile);
+  }
```

Comment:
There was an issue where tests would sometimes fail when running in parallel. 
One of the tests in the test suite changed the current working directory for **all tests**. When other tests required the current working directory to be something specific ran, this could cause them to fail.

My solution was to not actually change the current working directory. It does make the code somewhat more confusing though...

Reply:
Done!